### ISSUE TITLE
Change management for alt text validation

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#15944]
The change represented by epic #13103 and #15785

## Change description from the perspective of CMS users
1. CMS Editors, who upload a new image to either the Media Library or within a Content node will be shown errors if the alt text they add for their image:

- Is Over 150 characters long
- Contains a file extension (.png, .jpg, .jpeg, or .gif)
- Uses phrases like “image of”, “photo of”, “graphic of”, “picture of”.

These errors will be shown immediately upon focusing out of the alt text field, meaning that the editor will receive the feedback that something within their alt text must be corrected prior to trying to save the image.

2. CMS Editors, who edit an existing image, will be shown errors if the existing alt text for the image violates one of the following:

- Is Over 150 characters long
- Contains a file extension (.png, .jpg, .jpeg, or .gif)
- Uses phrases like “image of”, “photo of”, “graphic of”, “picture of”.

These errors will be shown even if the editor's original intention was not to edit the alt text. These errors will be shown when the editor attempts to save the image and must be corrected before the image will be allowed to be saved.

For example, if an editor edits an existing image in order to add a new crop ratio, but their alt text has happened to be 200 characters, when the editor tries to save the image, an error message will appear letting them know that their alt text cannot be more than 150 characters. The editor will need to adjust their alt text to fit the character count limit before being able to save the image.

## Change description from a CMS/platform and project goal perspective
CMS goal is to create a better alt text experience for Veterans visiting VA.gov. Lengthy alt text, filenames and certain phrases can all create a poor experience for Veterans who rely on screen readers when visiting the site. By putting validation in place within the CMS to prevent these types of errors from being included within the image alt text, we are working towards this goal of improving the alt text that is provided on VA.gov.

## Role(s) impacted by the change
- [ ] All CMS users
- [x] All CMS editors
- [ ] Vet Center editor
- [ ] VA Medical Center editor
- [ ] VA Outreach Hub editor
- [ ] VA Benefits Hub editor
- [ ] Campaign landing page editor
- [ ] VA Resources and Support editor
- [ ] VBA editor
- [ ] Supporting roles such as CMS account admin, Facilities or Public websites team member

## Success measures
Success will be measured by:

- Google sheet report
  - Prior to the new validations being launched, a report will be run to generate updated statistics regarding the current alt text. This will include the percentage of alt text that currently:
    - Is more than 150 characters
    - Contains filenames
    - Contains redundant phrases
  - Once the validations have been launched, the same statistics will be run again after a determined time period to see how they have changed.
    - We anticipate that no new images will:
      - Be longer than 150 characters
      - Contain filenames
      - Contain redundant phrases
    - We will also be looking at the existing alt text to see if editors updated anything existing based on the new validations. 
- Datadog dashboard metrics
  - The new validations will be set up within Datadog
  - From here we will be able to see which error is being triggered more often, least often, etc.
  - We can draw insights from this data and iterate on the validations if needed

## Tactics to be used (select all that apply)
- [x] Knowledge Base article(s)
https://prod.cms.va.gov/help/cms-basics/alternative-text
- [x] Content Style guidelines
https://design.va.gov/content-style-guide/alternative-text-for-images
- [x] New article (URL Pending)

## Pre-launch testing with users
- [ ] Yes
- [x] No

### Team
Please check the team(s) that will do this work.

- [x] `CMS Team`
- [ ] `Public Websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

**Comment by BerniXiongA6** [comment]
@laflannery Hope you don't mind that I'm adding this ticket to the current Sprint 99 only because Marcia and I have it on our radar to discuss a rollout strategy for Alt Text (which is likely to happen in the new year). Would make sense to have this ticket handy to provide @maortiz-27-80 and me with context, even if it doesn't done/finished by end of sprint. Thank you for putting this together! You're the best!

**Comment by laflannery** [comment]
@BerniXiongA6 Yes, I definitely think we should get this moving so we can get this shipped early Q1. This ticket is a little funny in the sense that there aren't ACs on this exact ticket but rather it's just an outline of a plan and I think it maybe would make sense if the 3 of us were to meet so we can figure out what tickets/actions we do need to take from this plan. CC: @maortiz-27-80

**Comment by maortiz-27-80** [comment]
Notes:

- Launched on 01/11
- Link to Product Outline
- Link to Rollout Strategy
- Ticket #16661 created for creation of datadog dashboard

Closing this ticket as complete. cc: @BerniXiongA6 @EWashb
